packages/services/schema/src/lib/federation-tag-extraction.ts

```diff
@@ -472,10 +353,44 @@ export function applyTagFilterOnSubgraphs<
     name: string;
   },
 >(subgraphs: Array<TType>, filter: Federation2SubgraphDocumentNodeByTagsFilter): Array<TType> {
-  const filteredSubgraphs = subgraphs.map(subgraph => ({
-    ...subgraph,
-    ...applyTagFilterToInaccessibleTransformOnSubgraphSchema(subgraph.typeDefs, filter),
-  }));
+  const specImpl = (resolveName: (name: string) => string) => (subgraph: TType) => {
+    const inaccessibleName = resolveName('@inaccessible');
+    const tagName = resolveName('@tag');
+    const filteredSubgraph = {
+      ...subgraph,
+      ...applyTagFilterToInaccessibleTransformOnSubgraphSchema(
+        subgraph.typeDefs,
+        filter,
+        inaccessibleName,
+        tagName,
+      ),
+    };
+    return filteredSubgraph;
+  };
+
+  // for every subgraph, get the spec and apply it.
+  // Current behavior is to treat fed1 and fed2 the same and support all versions.
+  const federationTagFilterSpec = new LinkableSpec('https://specs.apollo.dev/federation', {
+    [FEDERATION_V1]: specImpl,
+    v1_1: specImpl,
+    v2_9: specImpl,
```

Comment:
Currently, this configuration will support v1.0, v1.1, v2.0 - v2.9. If 2.10 gets released and used, then an error will be thrown because the spec does not support 2.10.

This is intentional because 2.10 may introduce a new feature that is not implemented. But I want to make sure that's okay with everyone since the current approach would assume it is still supported based on the name and therefore not throw.